IoT-Projects/ESP8266/Brushless Speed Controller/UBIDOTS_ESP8266_ESC-30A_BRUSHLESS_SPEED_CONTROLLER.ino

```diff
@@ -0,0 +1,68 @@
+/****************************************
+   Include Libraries
+ ****************************************/
+
+#include "UbidotsESPMQTT.h"
+#include <Servo.h>
+/****************************************
+   Define Constants
+ ****************************************/
+#define TOKEN "..........................." // Your Ubidots TOKEN
+#define WIFINAME "........." //Your SSID
+#define WIFIPASS "........." // Your Wifi Pass
+#define DEVICE_LABEL  "motor"  // Put here your Ubidots device label
+#define VARIABLE  "speed"  // Put here your Ubidots variable label
+#define MotorPin D5 //NodeMCU pin where the signal for the ESC comes out
+Servo ESC;
+float value=0; // To store incoming value.
+float MotorSpeed=0;
+
+Ubidots client(TOKEN);
+
+/****************************************
+   Auxiliar Functions
+ ****************************************/
+ 
+// cast from an array of chars to float value.
+float btof(byte * payload, unsigned int length) {
+    char * demo_ = (char *) malloc(sizeof(char) * 10);
+    for (int i = 0; i < length; i++) {
+        demo_[i] = payload[i];
+    }
+    return atof(demo_);
+}
+
+// Callback to handle subscription
+
+void callback(char* topic, byte* payload, unsigned int length) {
+    value = btof(payload, length);
+    value = map(value, 0, 100, 0, 180); //Map the 0-100 values from the slider to the 0-180 to use the servo lib.
+    ESC.write(value); //Send the value (PWM) to the ESC
+    Serial.println(value);
+    Serial.println(MotorSpeed);
+   
+}
+
+/****************************************
+   Main Functions
+ ****************************************/
+
+void setup() {
+    // put your setup code here, to run once:
```

Comment:
2 spaces instead of 4 spaces, please follow the style guide from google: https://google.github.io/styleguide/cppguide.html#Functions